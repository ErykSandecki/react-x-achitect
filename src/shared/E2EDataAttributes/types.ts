// types
import { E2EAttribute } from 'types/e2e';

export type E2EType = E2EAttribute;
export type E2EValue = number | string;
