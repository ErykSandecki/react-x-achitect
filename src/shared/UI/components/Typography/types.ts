// types
import { TTypographyProps } from './Typography';

export type TOmittedTypographyProps = Omit<TTypographyProps, 'fontType'>;
