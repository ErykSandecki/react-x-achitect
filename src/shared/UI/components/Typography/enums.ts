export enum TypographyColor {
  blue1 = 'blue1',
  green1 = '28a745',
  green2 = 'a7b45d',
  neutral1 = 'neutral1',
  neutral2 = 'neutral2',
  neutral3 = 'neutral3',
  neutral4 = 'neutral4',
  neutral5 = 'neutral5',
  orange1 = 'orange1',
  orange2 = 'orange2',
  pink1 = 'pink1',
  red1 = 'red1',
  skyBlue1 = 'skyBlue1',
  violet1 = 'violet1',
}

export enum TypographyFontStyle {
  italic = 'italic',
  normal = 'normal',
}

export enum TypographyFontType {
  h1 = 'h1',
  h2 = 'h2',
  h3 = 'h3',
  h4 = 'h4',
  h5 = 'h5',
  h6 = 'h6',
  p = 'p',
  small = 'small',
}

export enum TypographyFontWeight {
  blacks = 'blacks',
  bold = 'bold',
  extraBold = 'extra-bold',
  light = 'light',
  medium = 'medium',
  regular = 'regular',
  semiBold = 'semi-bold',
}
