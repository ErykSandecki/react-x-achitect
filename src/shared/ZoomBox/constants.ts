// types
import { T3DCoordinates } from 'types';

export const INITIAL_COORDINATES: T3DCoordinates = { x: 0, y: 0, z: 1 };
