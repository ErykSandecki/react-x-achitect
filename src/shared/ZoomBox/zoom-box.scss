.ZoomBox {
  cursor:
    url('assets/icons/cursors/default.png') 9 9,
    auto;

  &--move {
    cursor:
      url('assets/icons/cursors/move.png') 8 8,
      auto !important;
  }

  &--pressing,
  &--rmb {
    cursor:
      url('assets/icons/cursors/pressing.png') 8 8,
      auto !important;
  }

  &__zoom-content {
    transform-origin: 0 0;
  }
}
