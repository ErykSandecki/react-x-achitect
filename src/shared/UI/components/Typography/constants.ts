// others
import { COLORS } from 'constant/scss/variables/colors';

// types
import { Theme } from 'types/enums/theme';
import { TTypographyColorsMode } from './types';

export const TYPOGRAPHY_COLORS_MODE: TTypographyColorsMode = {
  link: {
    [Theme.dark]: COLORS.blue1Dark,
    [Theme.light]: COLORS.blue1Light,
  },
  neutral1: {
    [Theme.dark]: COLORS.neutral1Dark,
    [Theme.light]: COLORS.neutral1Light,
  },
  neutral2: {
    [Theme.dark]: COLORS.neutral2Dark,
    [Theme.light]: COLORS.neutral2Light,
  },
  neutral3: {
    [Theme.dark]: COLORS.neutral3Dark,
    [Theme.light]: COLORS.neutral3Light,
  },
  neutral4: {
    [Theme.dark]: COLORS.neutral4Dark,
    [Theme.light]: COLORS.neutral4Light,
  },
  neutral5: {
    [Theme.dark]: COLORS.neutral5Dark,
    [Theme.light]: COLORS.neutral5Light,
  },
} as const;
