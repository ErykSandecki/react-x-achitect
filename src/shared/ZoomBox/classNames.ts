import { mapValues } from 'lodash';

// others
import { CURSOR_STATES } from 'constant/constants';

// types
import { MouseMode } from 'types/enums/mouseMode';

export const className = 'ZoomBox';

export const classes = {
  className,
};

export const classNames = {
  [className]: {
    name: className,
    modificators: {
      pressing: `${className}__pressing`,
      ...CURSOR_STATES.reduce(
        (obj, key) => ({ ...obj, [key]: `${className}__${key}` }),
        {},
      ),
      ...mapValues(MouseMode, (mode) => `${className}__${mode}`),
    },
  },
  backgroundMask: `${className}__background-mask`,
  textureBlank: `${className}__texture-blank`,
  zoomContent: `${className}__zoom-content`,
};
