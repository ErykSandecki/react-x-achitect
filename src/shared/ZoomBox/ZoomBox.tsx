import { FC, ReactNode, RefObject, useRef } from 'react';

// components
import Box from '../UI/components/Box/Box';

// hooks
import { useTheme } from 'hooks';
import { useZoomBoxEvents } from './hooks/useZoomBoxEvents';

// others
import { className, classNames } from './classNames';

// styles
import styles from './zoom-box.scss';

// types
import { T3DCoordinates } from 'types';

export type TZoomBoxProps = {
  children: ReactNode;
  coordinates: T3DCoordinates;
  setCoordinates: (coordinates: T3DCoordinates) => void;
  zoomBoxRef: RefObject<HTMLDivElement>;
};

export const ZoomBox: FC<TZoomBoxProps> = ({
  children,
  coordinates,
  setCoordinates,
  zoomBoxRef,
}) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const zoomContentRef = useRef<HTMLDivElement>(null);
  const { cursorState, ...events } = useZoomBoxEvents(
    coordinates,
    setCoordinates,
    zoomBoxRef,
  );

  return (
    <Box
      classes={{
        className: cx(
          classNamesWithTheme[className].name,
          classNamesWithTheme[className].modificators[cursorState],
        ),
      }}
      ref={zoomBoxRef}
      sx={{ bg: 'neutral4', height: '100%', overflow: 'hidden' }}
      {...events}
    >
      <Box
        classes={{ className: cx(classNamesWithTheme.zoomContent) }}
        ref={zoomContentRef}
        sx={{ height: '100vh', position: 'relative' }}
        style={{
          transform: `translate(${coordinates.x}px, ${coordinates.y}px) scale(${coordinates.z})`,
        }}
      >
        {children}
      </Box>
    </Box>
  );
};

export default ZoomBox;
