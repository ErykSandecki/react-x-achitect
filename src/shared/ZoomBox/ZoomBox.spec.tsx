import { fireEvent, render } from '@testing-library/react';
import { noop } from 'lodash';

// components
import ZoomBox from './ZoomBox';

// others
import { BASE_3D } from './constants';

// types
import { E2EAttribute, MouseButton } from 'types';
import { MouseMode } from 'components/PageBuilder/enums';

// utils
import { getByE2EAttribute } from 'test';

const mockCallBack = jest.fn();
const ref = { current: null };

describe('ZoomBox snapshots', () => {
  it('should render Zoombox', () => {
    //before
    const { asFragment } = render(
      <ZoomBox
        alpha="100"
        backgroundColor="#ffffff"
        backgroundVissible
        coordinates={BASE_3D}
        mouseMode={MouseMode.default}
        onMouseDown={mockCallBack}
        onMouseMove={mockCallBack}
        onMouseUp={mockCallBack}
        setCoordinates={noop}
        zoomBoxRef={ref}
      >
        children
      </ZoomBox>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when mouse mode is move & cursor state is pressing', () => {
    //before
    const { asFragment, container } = render(
      <ZoomBox
        alpha="100"
        backgroundColor="#ffffff"
        backgroundVissible
        coordinates={BASE_3D}
        mouseMode={MouseMode.move}
        onMouseDown={mockCallBack}
        onMouseMove={mockCallBack}
        onMouseUp={mockCallBack}
        setCoordinates={noop}
        zoomBoxRef={ref}
      >
        children
      </ZoomBox>,
    );

    // action
    fireEvent.mouseDown(
      getByE2EAttribute(container, E2EAttribute.box, 'zoom-box'),
      {
        buttons: MouseButton.lmb,
      },
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
