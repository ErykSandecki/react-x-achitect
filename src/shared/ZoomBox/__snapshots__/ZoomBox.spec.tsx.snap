// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomBox snapshots should render Zoombox 1`] = `
<DocumentFragment>
  <div
    class="css-p683st ZoomBox ZoomBox--light ZoomBox__idle ZoomBox__idle--light ZoomBox__default ZoomBox__default--light"
    data-test-box="zoom-box"
  >
    <div
      class="css-q0izro ZoomBox__background-mask ZoomBox__background-mask--light"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); display: initial;"
    />
    <div
      class="css-q0izro ZoomBox__texture-blank ZoomBox__texture-blank--light"
      data-test-box=""
      style="transform: translate(0px, 0px) scale(1);"
    />
    <div
      class="css-qel697 ZoomBox__zoom-content ZoomBox__zoom-content--light"
      data-test-box=""
      style="transform: translate(0px, 0px) scale(1);"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`ZoomBox snapshots should render when mouse mode is move & cursor state is pressing 1`] = `
<DocumentFragment>
  <div
    class="css-p683st ZoomBox ZoomBox--light ZoomBox__lmb ZoomBox__lmb--light ZoomBox__move ZoomBox__move--light ZoomBox__pressing ZoomBox__pressing--light"
    data-test-box="zoom-box"
  >
    <div
      class="css-q0izro ZoomBox__background-mask ZoomBox__background-mask--light"
      data-test-box=""
      style="background-color: rgb(255, 255, 255); display: initial;"
    />
    <div
      class="css-q0izro ZoomBox__texture-blank ZoomBox__texture-blank--light"
      data-test-box=""
      style="transform: translate(0px, 0px) scale(1);"
    />
    <div
      class="css-qel697 ZoomBox__zoom-content ZoomBox__zoom-content--light"
      data-test-box=""
      style="transform: translate(0px, 0px) scale(1);"
    >
      children
    </div>
  </div>
</DocumentFragment>
`;
